ons-spark/spark-concepts/cache.md

```diff
@@ -0,0 +1,591 @@
+```
+# Using cache for more efficient coding
+
+Let's take a look at caching to speed up some processing. The scenario is that we want to read in some data and apply some cleaning and preprocessing. Then using this *cleansed* DataFrame we want to produce three tables as outputs for a publication.
+
+The data we will use are the animal rescue data set collected by the London Fire Brigade.
+
+## Preparing the data
+````{tabs}
+```{code-tab} py
+rescue = spark.read.csv(
+    config['rescue_path_csv'], 
+    header=True, inferSchema=True, 
+)
+```
+
+```{code-tab} r R
+
+rescue <- sparklyr::spark_read_csv(sc, config$rescue_path_csv, header=TRUE, infer_schema=TRUE) 
+
+```
+````
+The DataFrame is read into a single partition, which is fine because it's a small amount of data. However, we'll repartition to later demonstrate a feature of Spark.
+````{tabs}
+```{code-tab} py
+rescue = rescue.repartition(2)
+
+rescue = rescue.drop(
+    'WardCode', 
+    'BoroughCode', 
+    'Easting_m', 
+    'Northing_m', 
+    'Easting_rounded', 
+    'Northing_rounded'
+)
+
+
+rescue = (rescue
+          .withColumnRenamed("PumpCount", "EngineCount")
+          .withColumnRenamed("FinalDescription", "Description")
+          .withColumnRenamed("HourlyNotionalCost(£)", "HourlyCost")
+          .withColumnRenamed("IncidentNotionalCost(£)", "TotalCost")
+          .withColumnRenamed("OriginofCall", "OriginOfCall")
+          .withColumnRenamed("PumpHoursTotal", "JobHours")
+          .withColumnRenamed("AnimalGroupParent", "AnimalGroup")
+)
+
+
+rescue = rescue.withColumn(
+    "DateTimeOfCall", F.to_date(rescue.DateTimeOfCall, "dd/MM/yyyy")
+)
+
+
+rescue = rescue.withColumn(
+    "IncidentDuration", 
+    rescue.JobHours / rescue.EngineCount
+)
+
+
+rescue = rescue.na.drop(subset=["TotalCost", "IncidentDuration"])
+```
+
+```{code-tab} r R
+
+rescue <- rescue %>% sparklyr::sdf_repartition(2)
+
+rescue <- rescue %>% sparklyr::select(
+     - WardCode, 
+     - BoroughCode, 
+     - Easting_m, 
+     - Northing_m, 
+     - Easting_rounded, 
+     - Northing_rounded
+)
+
+
+rescue <- rescue %>% dplyr::rename(
+    EngineCount = PumpCount,
+    Description = FinalDescription,
+    HourlyCost = HourlyNotionalCostGBP,
+    TotalCost = IncidentNotionalCostGBP,
+    OriginOfCall = OriginofCall,
+    JobHours = PumpHoursTotal,
+    AnimalGroup = AnimalGroupParent
+)
+
+
+rescue <- rescue %>% sparklyr::mutate(
+    DateTimeOfCall = to_date(DateTimeOfCall, "dd/MM/yyyy")) 
+
+
+
+rescue <- rescue %>% 
+    sparklyr::mutate(IncidentDuration = JobHours / EngineCount)
+
+
+rescue <- rescue  %>% 
+      sparklyr::filter(!is.na(TotalCost) | !is.na(IncidentDuration))
+
+```
+````
+Note that the above cells executed very quickly. Remember this is because they're all transformations, which means Spark hasn't executed the code yet, take a look at the Spark UI if need convincing. So far Spark has created an execution plan but is waiting for an action to implement the plan.
```

Comment:
I think you're missing a word: take a look at the Spark UI if you need convincing. Also probably better to put a semicolon (;) over a comma there.